src/components/InputTime/InputTime.tsx

```diff
@@ -25,12 +29,115 @@ Desired features:
 */
 
 interface InputTimeProps extends Omit<InputProps, 'value' | 'max' | 'min'> {
+  max?: Date | string;
+  min?: Date | string;
+  value?: Date | string;
+}
+
+export const InputTime: React.FC<InputTimeProps> = ({
+  disabled: initDisabled,
+  forwardedRef,
+  max,
+  min,
+  value,
+  ...passedProps
+}) => {
+  let disabled = initDisabled;
+
+  if (value instanceof Date) {
+    const maxDate = max instanceof Date ? max : undefined;
+    const minDate = min instanceof Date ? min : undefined;
+    const now = new Date();
+
+    if (maxDate && maxDate < getStartOfDay(now)) {
+      disabled = true;
+    }
+
+    if (minDate && minDate > getEndOfDay(now)) {
+      disabled = true;
+    }
+
+    return (
+      <InputTimeDate
+        {...passedProps}
+        disabled={disabled}
+        max={maxDate}
+        min={minDate}
+        value={value}
+      />
+    );
+  }
+
+  const maxString = typeof max === 'string' ? max : undefined;
+  const minString = typeof min === 'string' ? min : undefined;
+  const valueString = typeof value === 'string' ? value : undefined;
+
+  return (
+    <InputTimeString
+      {...passedProps}
+      max={maxString}
+      min={minString}
+      value={valueString}
+    />
+  );
+};
+
+interface InputTimeDateProps extends Omit<InputProps, 'value' | 'max' | 'min'> {
+  max?: Date;
+  min?: Date;
+  value?: Date;
+}
+
+export const InputTimeDate: React.FC<InputTimeDateProps> = ({
+  max: maxDate,
+  min: minDate,
+  onChange,
+  value: valueDate,
+  ...passedProps
+}) => {
+  const now = new Date();
+
+  const max = useMemo(() => {
+    if (maxDate === undefined || maxDate === null) {
+      return '';
+    }
+
+    if (maxDate > getEndOfDay(now)) {
+      return '';
+    }
+
+    return getShortTimeString(maxDate.getHours(), maxDate.getMinutes());
+  }, [maxDate, now]);
```

Comment:
This effectively removes the need for `useMemo`. `now` will _always_ be a different value on each re-render so no memorization will ever happen. This may as well be a variable assignment or a normal function

Reply:
Crud, good call.
I'm not sure how often to generate the date.
Upside of always re-calculating: If you're on the form at midnight, and your date suddenly becomes invalid, we catch it.

But it seems like a lot of recalculating and date parsing for very little benefit.

What do you think? Should I consider getting `now` just once on mount?

Reply:
I think I would lean towards that.

However, it may make sense to floor to the nearest minute. That way if a few minutes pass during a restroom break or something the user can come back and it will be recalculated to the most up-to-date minute.